label_trans_selinux.go

```diff
@@ -49,35 +68,30 @@ func sendRequest(conn net.Conn, t requestType, data string) (string, error) {
 	}
 	defer f.Close()
 
-	if _, err := vectorio.WritevRaw(f.Fd(), v); err != nil {
+	writer := net.Buffers(v)
+	if _, err := writer.WriteTo(c); err != nil {
 		return "", fmt.Errorf("failed to write to mcstransd: %w", err)
 	}
 
 	var uintsize uint32
-	elemsize := uint64(unsafe.Sizeof(uintsize))
-
-	header := []syscall.Iovec{
-		{Len: elemsize}, // function
-		{Len: elemsize}, // response length
-		{Len: elemsize}, // return value
-	}
 
-	if _, err := vectorio.ReadvRaw(f.Fd(), header); err != nil {
+	var hdr [unsafe.Sizeof(uintsize) * 3]byte
+	_, err = c.Read(hdr[0:])
+	if err != nil {
 		return "", fmt.Errorf("failed to read from mcstransd: %w", err)
 	}
 
-	respvec := []syscall.Iovec{{Len: uint64(*header[1].Base)}}
-	_, err = vectorio.ReadvRaw(f.Fd(), respvec)
-	if err != nil {
-		return "", fmt.Errorf("failed to read from response: %w", err)
-	}
+	//function := nativeEndian.Uint32(hdr[0:4]) // unused
+	responselen := nativeEndian.Uint32(hdr[4:8])
+	//returncode := nativeEndian.Uint32(hdr[8:12]) // unused
```

Comment:
Can you remove these comments?

Reply:
I left them there on purpose so that if/when we need those fields whoever is doing that doesn't have to try to figure out the format. 

I could add them to a debug printf so they aren't unused anymore, or change the comment to prose form...

Reply:
That makes sense.

How about:

```
// the hdr buffer contains the following structure:
// function: 0-4
// response length: 4-8
// return code: 8-12
responselen := nativeEndian.Uint32(hdr[4:8])
```